networkdb/delegate.go

```diff
@@ -165,16 +165,19 @@ func (nDB *NetworkDB) handleTableEvent(tEvent *TableEvent) bool {
 		}
 	}
 	nDB.RUnlock()
+
 	if !ok || network.leaving || !nodePresent {
 		// I'm out of the network OR the event owner is not anymore part of the network so do not propagate
 		return false
 	}
 
+	nDB.Lock()
```

Comment:
how about `defer nDB.Unlock()` right after `nDB.Lock()` here? That way no need to `unlock` in the exit scenario down below and again at the very end

Reply:
was simply trying to avoid keeping the lock for more than was necessary like the dispatch of the event. For symmetry with the other methods after the entry is inserted into the tree you can release the lock and handle whatever operation is following, like sending the notification to other nodes like the case of Create/Update/Delete or like here dispatching the event to the app

Reply:
sounds good wrt symmetry.